File: backend/save/save_rebuilt_landing_2d.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
from __future__ import print_function
import sys, json, os, time, io


def respond(payload):
    print(json.dumps(payload, ensure_ascii=False), end='')


def ensure_dir(path):
    if not os.path.isdir(path):
        os.makedirs(path)


def main():
    raw = sys.stdin.read()
    try:
        data = json.loads(raw)
    except Exception:
        respond({'ok': False, 'error': 'invalid json'})
        return
    base = os.path.join(os.path.dirname(os.path.dirname(__file__)), '海缆数据', '重构的数据')
    ensure_dir(base)
    ap_items = []
    if isinstance(data, dict):
        if isinstance(data.get('ap'), list):
            ap_items = data['ap']
        elif isinstance(data.get('items'), list):
            ap_items = data['items']
    elif isinstance(data, list):
        ap_items = data
    standard_items = []
    if isinstance(data, dict) and isinstance(data.get('standard'), list):
        standard_items = data['standard']
    if not standard_items:
        standard_items = ap_items
    saved_at = int(time.time() * 1000)

    def write_file(path, items):
        payload = {'version': 1, 'savedAt': saved_at, 'items': list(items)}
        with io.open(path, 'w', encoding='utf-8') as f:
            json.dump(payload, f, ensure_ascii=False)

    ap_path = os.path.join(base, '2d亚太中心登陆站数据.json')
    std_path = os.path.join(base, '2d标准地图登陆站数据.json')
    write_file(ap_path, ap_items)
    write_file(std_path, standard_items)
    respond({'ok': True, 'paths': {'ap': ap_path, 'standard': std_path}})


if __name__ == '__main__':
    main()

File: backend/save/save_rebuilt_landing_2d.php
<?php
header('Content-Type: application/json; charset=utf-8');
try {
    $raw = file_get_contents('php://input');
    if ($raw === false || strlen($raw) === 0) {
        http_response_code(400);
        echo json_encode([ 'ok' => false, 'error' => 'empty body', 'engine' => 'php' ], JSON_UNESCAPED_UNICODE);
        exit;
    }
    $decoded = json_decode($raw, true);
    if ($decoded === null) {
        http_response_code(400);
        echo json_encode([ 'ok' => false, 'error' => 'invalid json', 'engine' => 'php' ], JSON_UNESCAPED_UNICODE);
        exit;
    }
    $baseDir = dirname(__DIR__, 2) . '/seacable_data/重构的数据';
    if (!is_dir($baseDir)) {
        if (!mkdir($baseDir, 0775, true) && !is_dir($baseDir)) {
            http_response_code(500);
            echo json_encode([ 'ok' => false, 'error' => 'mkdir failed', 'engine' => 'php' ], JSON_UNESCAPED_UNICODE);
            exit;
        }
    }
    $apItems = [];
    if (isset($decoded['ap']) && is_array($decoded['ap'])) {
        $apItems = $decoded['ap'];
    } elseif (isset($decoded['items']) && is_array($decoded['items'])) {
        $apItems = $decoded['items'];
    } elseif (is_array($decoded)) {
        $apItems = $decoded;
    }
    $standardItems = [];
    if (isset($decoded['standard']) && is_array($decoded['standard'])) {
        $standardItems = $decoded['standard'];
    }
    if (!$standardItems) {
        $standardItems = $apItems;
    }
    $savedAt = (int) round(microtime(true) * 1000);
    $writeFile = function ($filePath, $items) use ($savedAt) {
        $payload = [ 'version' => 1, 'savedAt' => $savedAt, 'items' => array_values($items) ];
        $json = json_encode($payload, JSON_UNESCAPED_UNICODE | JSON_UNESCAPED_SLASHES);
        if ($json === false) {
            throw new RuntimeException('encode failed');
        }
        $ok = file_put_contents($filePath, $json);
        if ($ok === false) {
            throw new RuntimeException('write failed');
        }
    };
    $apPath = $baseDir . '/2d亚太中心登陆站数据.json';
    $stdPath = $baseDir . '/2d标准地图登陆站数据.json';
    $writeFile($apPath, $apItems);
    $writeFile($stdPath, $standardItems);
    echo json_encode([ 'ok' => true, 'paths' => [ 'ap' => $apPath, 'standard' => $stdPath ], 'engine' => 'php' ], JSON_UNESCAPED_UNICODE);
} catch (Throwable $e) {
    http_response_code(500);
    echo json_encode([ 'ok' => false, 'error' => $e->getMessage(), 'engine' => 'php' ], JSON_UNESCAPED_UNICODE);
}
